src/ray/rpc/grpc_server.cc

```diff
@@ -48,7 +55,7 @@ void GrpcServer::PollEventsFromCompletionQueue() {
         // incoming request.
         server_call->GetFactory().CreateCall();
         server_call->SetState(ServerCallState::PROCESSING);
-        main_service_.post([server_call] { server_call->HandleRequest(); });
+        server_call->GetIOService().post([server_call] { server_call->HandleRequest(); });
```

Comment:
Maybe do `post` inside `HandleRequest`? Then the `ServerCall` doesn't need to expose `GetIOService`.

Reply:
makes sense. revised